lib/TTT/game.ex

```diff
@@ -1,38 +1,56 @@
 defmodule Game do
-  def init(token \\ "X") do
-    %State{player: token}
+  def init(player_token \\ "X") do
+    %State{player: player_token}
   end
 
+  def select_mode(mode) when mode == "1", do: :human_v_human
+  def select_mode(mode) when mode == "2", do: :human_v_computer
+
   def over(state) do
-    Output.display_board(state)
-    Output.end_of_game(state)
+    Messages.display_board(state)
+    Messages.end_of_game(state)
+  end
+
+  def play(state, deps, mode) when mode == :human_v_human do
+    deps.messages.display_board(state)
+
+    move(state, deps)
+    |> check_status(deps, :human_v_human)
   end
 
-  def play(state, deps) do
-    deps.output.display_board(state)
+  def play(state, deps, mode) when mode == :human_v_computer do
+    deps.messages.display_board(state)
 
-    new_state =
-      human_move(state, deps)
-      |> deps.output.display_board()
-      |> computer_move()
-      |> deps.output.display_message(:opponent_move)
+    move(state, deps)
+    |> computer_move()
+    |> check_status(deps, :human_v_computer)
+  end
```

Comment:
Are overloaded functions an elixir convention? I think they might be but wouldn't mind having a discussion about it either way (here or in person). Seems like having two `play` functions implies that the way to add a new player configuration is to write another `play` function. Might be nicer to work with (here's where the discussion comes in) if the same `play` could handle any player configuration.

Reply:
From what I understand overloaded functions are sometimes convention. I originally had the two game modes pivoting in a case statement within one `play` function, but then separated them into two functions that used guard clauses. I do agree with you that it would be nicer to have one function that handled any player configuration, but I couldn't figure out how to do it without some form of conditional logic

Reply:
Overloading in Elixir _is_ a convention, it uses pattern matching under the hood to infer the correct function.

But!  I definitely agree.  This specific case it seems like the only difference is that `computer_move()` is inserted in the pipeline.  I can second your intuition @davetorre  that it would probably be worth it to explore other approaches here if only for having that conversation.